lib/server/routes/reports.js

```diff
@@ -73,17 +152,108 @@ ReportsRouter.prototype.createExchangeReport = function(req, res, next) {
   });
 };
 
+/**
+ * Creates an exchange report
+ * @param {http.IncomingMessage} req
+ * @param {http.ServerResponse} res
+ * @param {Function} next
+ */
+ReportsRouter.prototype.createExchangeReport = function(req, res, next) {
+  const token = req.body.token;
+
+  /* jshint maxstatements: 30, maxcomplexity: 15 */
+  this.storage.models.StorageEvent.findOne({token: token}, (err, event) => {
+    if (err) {
+      return next(new errors.InternalError(err.message));
+    }
+
+    if (!event) {
+      return next(new errors.NotFoundError('Storage event not found'));
+    }
+
+    // TODO: Check that event isn't closed for reports
+
+    const report = {
+      exchangeStart: req.body.exchangeStart,
+      exchangeEnd: req.body.exchangeEnd,
+      exchangeResultCode: req.body.exchangeResultCode,
+      exchangeResultMessage: req.body.exchangeResultMessage
+    };
+
+    if (!this.validateExchangeReport(report)) {
+      return next(new errors.BadRequestError('Invalid exchange report'));
+    }
+
+    const isClientReport = req.user ?
+          (event.client === req.user.id) : req.farmerNodeID ?
```

Comment:
I believe this needs to compare `event.client` to `req.user._id` rather than `req.user.id`

Reply:
Okay, what the problem that comes up?